tools/server/webui/src/utils/storage.ts

```diff
@@ -192,15 +192,10 @@ const StorageUtils = {
   },
 
   // manage config
-  getConfig(): typeof CONFIG_DEFAULT {
-    const savedVal = JSON.parse(localStorage.getItem('config') || '{}');
-    // to prevent breaking changes in the future, we always provide default value for missing keys
-    return {
-      ...CONFIG_DEFAULT,
-      ...savedVal,
-    };
+  getConfig(): AppConfig {
+    return JSON.parse(localStorage.getItem('config') || '{}') as AppConfig;
```

Comment:
This can be a weak typing. Basically the `getConfig()` is guaranteed to return `AppConfig`, but now there is a chance it will return empty object `{}`, which may cause undefined value or even crash the whole app if some code tries to read keys from that empty object.